reactor-core/src/test/java/reactor/core/publisher/MonoErrorSuppliedTest.java

```diff
@@ -112,4 +113,17 @@ public void lazilyEvaluatedCall() {
 
 		assertThat(count).as("after call").hasValue(1);
 	}
+
+	@Test
+	public void supplierMethod() {
+		Mono<Object> error = Mono.error(illegalStateExceptionSupplier());
+
+		assertThatExceptionOfType(IllegalStateException.class)
```

Comment:
prefer using the `StepVerifier` rather than `block`. As state above, you could use `verifyErrorSatisfies` for a similarly complete assertion:

```java
.verifyErrorSatisfies(e -> assertThat(e).isInstanceOf(IllegalStateException.class)
		                                .hasMessage("boom"))
```